From 5e6017e3455a37700867c20f4a6ad3ebd85e397f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Thu, 23 Dec 2021 14:14:59 +0100
Subject: [PATCH 1/9] user32: Update rawinput device list in WM_DEVICECHANGE
 callback.

The SDL library calls GetRawInputDeviceList quickly after it has
detected a new device, from a WM_DEVICECHANGE notification.

It uses rawinput device information to find out whether it is an XInput
device, when its name contains "IG_", or if the device should be opened
through DInput.

If the device is missing from the rawinput list, it considers it as a
DInput device, and may end up with the same device used twice.
---
 dlls/user32/input.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/user32/input.c b/dlls/user32/input.c
index 11111111111..11111111111 100644
--- a/dlls/user32/input.c
+++ b/dlls/user32/input.c
@@ -532,6 +532,7 @@ BOOL WINAPI UnloadKeyboardLayout( HKL layout )
 
 static DWORD CALLBACK devnotify_window_callback(HANDLE handle, DWORD flags, DEV_BROADCAST_HDR *header)
 {
+    rawinput_update_device_list();
     SendMessageTimeoutW(handle, WM_DEVICECHANGE, flags, (LPARAM)header, SMTO_ABORTIFHUNG, 2000, NULL);
     return 0;
 }
-- 
2.36.1

